lib/core/storages/memory.ts

```diff
@@ -18,3 +18,107 @@ export class MemoryStorage implements IStorage {
     return promise;
   }
 }
+
+export class IndexedDbStorage implements IStorage {
```

Comment:
Can we move this to a new file under storages?

Reply:
done